src/main/java/core/basesyntax/service/impl/FruitTransactionMapper.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.enums.Operation;
+import core.basesyntax.service.Mapper;
+
+public class FruitTransactionMapper implements Mapper<FruitTransaction, String> {
+    private static final String DELIMITER = ",";
+    private static int OPERATION_INDEX = 0;
+    private static int FRUIT_INDEX = 1;
+    private static int QUANTITY_INDEX = 2;
+
+    @Override
+    public FruitTransaction toObject(String string) {
+        String[] split = string.split(DELIMITER);
+        validateSplitLength(split, string);
+        Operation operation = Operation.fromCode(split[OPERATION_INDEX]);
+        String fruit = split[FRUIT_INDEX];
+        int quantity = parseQuantity(split[QUANTITY_INDEX]);
+        return new FruitTransaction(operation, fruit, quantity);
```

Comment:
Consider encapsulating the parsing logic into a separate method to follow the Single Responsibility Principle. The `toObject` method should only be responsible for mapping, while parsing can be done in another method.